XenAdmin/Wizards/PatchingWizard/PatchingWizard_PrecheckPage.cs

```diff
@@ -601,12 +612,9 @@ private void buttonResolveAll_Click(object sender, EventArgs e)
                         actions.Add(action);
                 }
             }
-            foreach (var asyncAction in actions)
-            {
-                _progressDialog = new ActionProgressDialog(asyncAction, ProgressBarStyle.Blocks);
-                _progressDialog.ShowDialog(this);
-                Program.Invoke(Program.MainWindow, RefreshRechecks);
-            }
+            var multipleAction = new CrossConnectionParallelAction(Messages.PATCHINGWIZARD_PRECHECKPAGE_RESOLVING_ALL, "", "", actions);
+            _progressDialog = new ActionProgressDialog(multipleAction, ProgressBarStyle.Blocks);

```

Comment:
We should consider not having a separate modal dialog, but putting it inline on the wizard. We already have a progress bar on that page: we could move it to the bottom and reuse it.